WIP: Atoms Maven Mojo

stbischof opened the pull request.

stbischof committed:
initial mvn substrate plugin

stbischof committed:
use incremental index, fix ressource list

stbischof committed:
build atomos.substrate.jar and add to native-image classpath

stbischof committed:
create a proper jar file

stbischof committed:
close jarFile

tjwatson commented:
I see this is now building correctly.  Is there something you would like me to review or is this still WIP?

stbischof commented:
Still wip! Questions on gitter

stbischof committed:
add atomos to graal_resource_config.json

stbischof committed:
store graal_resource_config inside the jar
META-INF/native-image/graal_resource_config.json

stbischof committed:
fix resource_config.json name, aditional res_conf via pom config

stbischof committed:
handle proxy_config in pom

stbischof committed:
add dynProxyConfig to pom configuration

stbischof committed:
use GRAAL_HOME env var

stbischof committed:
use env corect

tjwatson commented on atomos.tests/atomos.tests.testbundles/pom.xml:
```diff
@@ -8,7 +8,7 @@
     <packaging>pom</packaging>
     <artifactId>org.apache.felix.atomos.tests.testbundles</artifactId>
-    <name>atomos.tests.testbundles</name>
+    <name>org.apache.felix.atomos.tests.testbundles</name>
```
Not sure why you changed the name here.  I would prefer to omit the org.apache.felix prefix from the name to make it smaller.

tjwatson reviewed (commented):
Initial comments only, with no real review ...

There are some files in the commits that I am sure should not be:

`/org.apache.felix.atomos.examples.substrate.maven.equinox`
all of the files under `/reports`

tjwatson committed:
Delete stuff that shouldn't be here

tjwatson committed:
Hacked up fixes

The fixes here should be reworked into the overall structure of the
code.  I only did quick fixes here that probably need more work to make
them good.

- quick fix to resources json to just include everything under atomos/
folder for resources
- fix up atomos.substrate.jar to not have entries begin  with '/'
- avoid adding JARs with no BSN to the bundle index file
- default to version 0.0 if there is no Bundle-Version header
- stick all bundle resources under atomos/ folder

stbischof committed:
working gogo and felix, with external reflect_config.

stbischof committed:
refactoring. split reflectConfig.json

stbischof committed:
refactor

stbischof committed:
merge

stbischof committed:
upate tu current api

stbischof committed:
test
